/*
*  This file is part of aasdk library project.
*  Copyright (C) 2018 f1x.studio (Michal Szwaj)
*
*  aasdk is free software: you can redistribute it and/or modify
*  it under the terms of the GNU General Public License as published by
*  the Free Software Foundation; either version 3 of the License, or
*  (at your option) any later version.

*  aasdk is distributed in the hope that it will be useful,
*  but WITHOUT ANY WARRANTY; without even the implied warranty of
*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
*  GNU General Public License for more details.
*
*  You should have received a copy of the GNU General Public License
*  along with aasdk. If not, see <http://www.gnu.org/licenses/>.
*/

syntax="proto2";

import "AVStreamTypeEnum.proto";
import "AudioTypeEnum.proto";
import "AudioConfigData.proto";
import "VideoConfigData.proto";
import "DisplayTypeEnum.proto";
import "VideoFocusReasonEnum.proto";
package oaa.proto.data;

message AVChannel
{
    required enums.AVStreamType.Enum stream_type = 1;
    optional enums.AudioType.Enum audio_type = 2;
    repeated AudioConfig audio_configs = 3;
    repeated VideoConfig video_configs = 4;
    optional bool available_while_in_call = 5;  // APK field 5 is uint32 (not bool). Not sent; omitting is safe.

    // Fields 6-9 discovered via APK decompilation (vys.java, AA v16.1).
    optional uint32 channel_id = 6;                             // Display/channel identifier (assigned by phone)
    optional enums.DisplayType.Enum display_type = 7;           // MAIN, CLUSTER, AUXILIARY
    optional uint32 keycode = 8;                                // Android keycode or AA-specific (65536+)
    optional enums.VideoFocusReason.Enum focus_reason = 9;      // Why video focus was set
}
